src/task2e.kt

```diff
@@ -0,0 +1,154 @@
+
+    val numbers = listOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
+    val squaredEvenNumbers = numbers.filter { it % 2 == 0 }.map { it * it }
+    println(squaredEvenNumbers)
+
+    /**
+     * Задание 2
+     * Создайте список строк [“один”, “два”, “три”, “четыре”, “пять”]. Используйте map чтобы получить длину каждого элемента списка.
+     */
+
+    val stringNumbers = listOf("один", "два", "три", "четыре", "пять")
+    val length = stringNumbers.map { it.length }
+    println(length)
+
+    /**
+     * Задание 3
+     * Напишите функцию высшего порядка, которая принимает два параметра: целое число и функцию, принимающую целое число и возвращающую строку. Эта функция должна преобразовывать число в строку с помощью переданной функции и возвращать результат.
+     */
+
+    val mapNumberWithString: (Int) -> String = { n ->
+        mapOf(
+            0 to "ноль",
+            1 to "один",
+            2 to "два",
+            3 to "три",
+            4 to "четыре",
+            5 to "пять",
+            6 to "шесть",
+            7 to "семь",
+            8 to "восемь",
+            9 to "девять"
+        )
+            .getOrDefault(n, "не определено")
+    }
+
+    fun numberToString(n: Int, operation: (Int) -> String): String {
+        return operation(n)
+    }
+    println(numberToString(5, mapNumberWithString))
+    println(numberToString(10, mapNumberWithString))
+
+    /**
+     * Задание 4
+     * Создайте лямбда-функцию, которая принимает число и возвращает его квадрат. Используйте эту лямбда-функцию в другой функции, которая принимает число и возвращает строку, состоящую из числа и его квадрата (например, “5 и его квадрат равен 25”).
+     */
+
+    val square: (Int) -> Int = { n -> n * n }
+    fun squareToString(n: Int, operation: (Int) -> Int): String {
+        val result = operation(n)
+        return "$n и его квадрат равен $result"
+    }
+    println(squareToString(12, square))
+
+    /**
+     * Задание 5
+     * Создайте объект класса Person со свойствами name и age. Создайте список из нескольких таких объектов. Используйте filter и map чтобы получить список имен тех людей, которым больше 18 лет.
+     */
+
+    class Person(val name: String, val age: Int)
+
+    val persons =
+        listOf(
+            Person("Tom", 23),
+            Person("Ann", 2),
+            Person("Mary", 17),
+            Person("Bill", 84)
+        )
+    val adultsPersons = persons.filter { it.age >= 18 }.map { it.name }
+    println(adultsPersons)
+
+    /**
+     * Задание 6
+     * Используя ленивые операции, создайте последовательность чисел от 1 до 1000, затем используйте filter чтобы оставить только числа, делящиеся на 5, затем map чтобы умножить каждое число на 2, и take чтобы взять первые 20 элементов этой последовательности.
+     */
+
+    val sequence = sequence {
+        var num = 1
+        while (true) {
+            yield(num++)
+        }
+    }.filter {
+        it % 5 == 0
+    }
+        .map {
+            it * 5
+        }
```

Comment:
```suggestion
    val sequence = sequence {
        var num = 1
        while (true) {
            yield(num++)
        }
    }
    	.filter { it % 5 == 0 }
        .map { it * 5 }
```
обычно короткие лямбды стараются так форматировать, так сильно читаемее